var/spack/repos/builtin/packages/warpx/package.py

```diff
@@ -24,21 +24,32 @@ class Warpx(MakefilePackage):
 
     variant('dims',
             default='3',
-            values=('2', '3'),
+            values=('2', '3', 'rz'),
             multi=False,
             description='Number of spatial dimensions')
-
+    variant('backend',
+            default='openmp',
+            values=('openmp', 'cuda', 'hip'),
+            multi=True,
+            description='Programming model for compute kernels')
+    variant('mpi', default=True, description='Enable MPI support')
     variant('psatd', default=False, description='Enable PSATD solver')
     variant('debug', default=False, description='Enable debugging features')
     variant('tprof', default=True, description='Enable tiny profiling features')
-    variant('openmp', default=True, description='Enable OpenMP features')
     variant('openpmd', default=True, description='Enable openPMD I/O')
+    variant('ascent', default=False, description='Enable Ascent in situ vis')
 
-    depends_on('mpi')
+    depends_on('cuda', when='backend=cuda')
+    depends_on('mpi', when='+mpi')
     depends_on('fftw@3:', when='+psatd')
+    depends_on('fftw +mpi', when='+psatd +mpi')
     depends_on('pkgconfig', type='build', when='+openpmd')
-    depends_on('python', type='build')  # FIXME upstream
-    depends_on('openpmd-api@0.11.0:,dev +mpi', when='+openpmd')
+    depends_on('python', type='build')  # AMReX' build system info
+    depends_on('openpmd-api@0.11.0:,dev', when='+openpmd')
+    depends_on('openpmd-api +mpi', when='+openpmd +mpi')
+    depends_on('ascent', when='+ascent')
+    depends_on('ascent +cuda', when='+ascent backend=cuda')
+    depends_on('ascent +mpi ^conduit~hdf5', when='+ascent +mpi')
```

Comment:
note @cyrush: the fixture of `conduit` to only accept non-MPI HDF5 variants makes it impossible to concretize ascent in default variants with an app that also provides parallel HDF5 I/O capabilities.

Reply:
-> #15500